### PR TITLE
Make use of 'pairing_batch' for added efficiency

### PR DESCRIPTION
**Pull request by jwasinger**
Also adds a test case for doing a pairing check with multiple points.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -162,9 +162,7 @@ pub fn bn128_pairing(input: &[u8], output: &mut [u8; 32]) -> Result<(), Error> {
             vals.push((a, b));
         }
 
-        let mul = vals
-            .into_iter()
-            .fold(Gt::one(), |s, (a, b)| s * pairing(a, b));
+        let mul = pairing_batch(&vals);
```
Comment (axic) [inline src/lib.rs:165]: Do you want to upstream this to parity also?

**Review by axic — CHANGES_REQUESTED** [review]
Can you split this into two commits? First push the test and have the CI finish. Then the change. This way we can verify that the old code also passed the test and this is not a change in behaviour but an optimisation.

File: src/lib.rs (hunk 1 above)
Comment (jwasinger) [reply src/lib.rs:165]: they already use it https://github.com/paritytech/parity-ethereum/pull/10765

**Comment by jwasinger** [comment]
Sure.

File: src/lib.rs (hunk 1 above)
Comment (axic) [reply src/lib.rs:165]: Ah ok! If you said it was from upstream i wouldn't have had any questions :)

**Review by axic — APPROVED** [review]

**Pull request merged**